composer.lock

```diff
@@ -0,0 +1,2612 @@
+{
```

Comment:
Do we really want to lock this here? Wouldn't we then miss if something break on newer library versions?

Reply:
Hi Sebastian,

as far as I know, it is common knowledge these days to actually include the `lockfiles` into the repository. That way, the version pinning is more solid and other subsystems like dependency caching can build upon it, like the excellent `actions/cache` GHA cache action [1].

I have not been aware about those details until recently but now I am an avid user of this, so we included `poetry.lock` and `yarn.lock` files throughout many repositories we are maintaining (like [2-5]) and we never looked back.

So, it obviously becomes the obligation of the package/repository maintainer to have a sound `lockfile` in place. Please let me know if you have further questions.

> Wouldn't we then miss if something break on newer library versions?

The purpose **actually is** that nothing ever breaks when everyone will adhere to the [semver specs](https://semver.org/), even if the package installers don't honor those `lockfile`s themselves. Let us have an example here from the Python world: When I designate a caret-style dependency on the `click` library within the project's `pyproject.toml` like `click = "^7.1.2"`, the sdist packaging step will render `click>=7.1.2,<8.0` to be put into the tarfile's `setup.py` to be uploaded to PyPI.

So, those mechanisms create a package pinning which automatically adds an upper boundary on the next major/minor revision, depending on the circumstances. That is actually a perfect [DWIM](https://en.wikipedia.org/wiki/DWIM) implementation. I haven't investigated thoroughly with PHP/composer, but I believe it will be the same.

In turn, it will be the obligation of the package maintainer to unlock newer major/minor dependency versions to the user of the package. However, it will perfectly follow updates on the minor/patch level of those.

With kind regards,
Andreas.

[1] https://github.com/actions/cache#whats-new
[2] https://github.com/earthobservations/wetterdienst
[3] https://github.com/panodata/grafana-pandas-datasource
[4] https://github.com/panodata/grafana-map-panel
[5] https://github.com/daq-tools/wireviz-web

Reply:
Hi again,

the article "Lockfiles should be committed on all projects" [1] explains it in much better terms than I have been able to do it.

With kind regards,
Andreas.

[1] https://classic.yarnpkg.com/blog/2016/11/24/lockfiles-for-all/